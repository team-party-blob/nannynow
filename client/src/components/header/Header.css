img {
  width: 8em;
}

/*Mobile Header */
@media (max-width: 650px) {
  header {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;

    img {
      display: none;
    }

    #hamburgerLabel {
      position: absolute;
      opacity: 0;
    }

    #hamburgerInput {
      width: 40px;
      height: 32px;
      position: fixed;
      top: 5px;
      left: 5px;
      opacity: 0;
      z-index: 9999999;
      &:hover {
        cursor: pointer;
      }
    }
    #hamburgerSpan {
      background: darkgrey;
      width: 33px;
      min-height: 4px;
      height: 4px;
      display: block;
      border-radius: 2px;
      margin-top: 20px;
      margin-left: 10px;
      transition: 0.5s;
      position: fixed;
      left: 0px;
      &::before, &::after {
        transition: 0.5s;
        border-radius: 2px;
        display: block;
        height: 4px;
        background: darkgrey;
        content: '';
        display: block;
        position: relative;
      }
      &:before {
          top:-8px;
      }
      &:after {
          top: 4px;
      }
    }

    input:checked ~ #hamburgerSpan::before {
      transform: rotate(45deg) translate(0px, -1px);
      top: 3px;
    }

    input:checked ~ #hamburgerSpan::after {
      transform: rotate(-45deg) translate(0px, -1px);
      top: -1px;
    }

    input:checked ~ #hamburgerSpan {
      background: rgba(0,0,0,0)
    }

    input:checked ~ #nav {
      transform: none;
      opacity: 1;
    }

    #nav {
      width: 85%;
      display: flex;
      flex-direction: column;
      opacity: 0;
      margin: 10px 0px 0px 10px;
      position: fixed;
      top: 2em;
      left: 1em;
      z-index: 999999;
      transform-origin: 0% 0%;
      transform: translate(-100%, 0);
      transition: transform 0.5s cubic-bezier(0.77,0.2,0.05,1.0);


      .link {
        text-align: center;
        padding: 10px;
        margin: 5px;
        background-color: #f96a54;
        color: black;
        text-decoration: none;
      }
    }
  }
}

/* Desktop Header */
@media (min-width: 651px) {
  header {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    width: 100%;
    // position: fixed;
    align-items: center;

    #hamburgerLabel {
      display: none;
    }

    #hamburgerInput {
      display: none;
    }

    img {
      padding-left: 5vw;
    }


    nav {
      .link {
        padding: 10px;
        margin: 20px;
        background-color: #f96a54;
        opacity: 0.8;
        color: white;
        text-decoration: none;
        &:hover {
          opacity: 1;
          cursor: pointer;
        }
      }
    }
  }
}
