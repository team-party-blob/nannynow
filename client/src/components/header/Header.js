import React from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';
import styles from './Header.css';
import { logo } from '../assets/imageUrl';

export default function Header() {
  return (
    <header>
      <img src={logo}/>


      <label id={styles.hamburgerLabel}>This is the hamburger input.</label>
      <input type="checkbox" id={styles.hamburgerInput} />
      <span id={styles.hamburgerSpan}></span>

      <nav id={styles.nav}>
        <Link className={styles.link} to={ROUTES.HOME.linkTo()}>Dashboard</Link>
        <Link className={styles.link} to={ROUTES.HOME.linkTo()}>Edit Profile</Link>
        <Link className={styles.link} to={ROUTES.HOME.linkTo()}>Log In</Link>
        <Link className={styles.link} to={ROUTES.HOME.linkTo()}>Log Out</Link>
      </nav>
    </header>
  );
}
