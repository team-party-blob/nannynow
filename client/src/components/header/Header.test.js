import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

jest.mock('../../assets/nannyNowLogo.png');

describe('Header component', () => {

  it('renders a Header component', () => {
    const session = { _id: '1234' };
    const wrapper = shallow(<Header session={session}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
