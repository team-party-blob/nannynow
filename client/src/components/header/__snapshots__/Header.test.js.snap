// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders a Header component 1`] = `
<header>
  <img
    src=""
  />
  <div>
    <label
      id="hamburgerLabel"
    >
      This is the hamburger input.
    </label>
    <input
      id="hamburgerInput"
      type="checkbox"
    />
    <span
      id="hamburgerSpan"
    />
    <nav
      id="nav"
    >
      <Link
        className="link"
        replace={false}
        to="/about"
      >
        About Us
      </Link>
      <Link
        className="link"
        replace={false}
        to="/dashboard/1234"
      >
        Dashboard
      </Link>
      <Link
        className="link"
        replace={false}
        to="/profile/1234"
      >
        Edit Profile
      </Link>
      <Link
        className="link"
        replace={false}
        to="/signup"
      >
        Log Out
      </Link>
    </nav>
  </div>
</header>
`;
